src/core/LiterallyCanvas.coffee

```diff
@@ -273,8 +281,12 @@ module.exports = class LiterallyCanvas
           @clipped (=>
             @transformed (=>
               for shape in @_shapesInProgress
-                renderShapeToContext(
-                  @ctx, shape, {@bufferCtx, shouldOnlyDrawLatest: true})
+                shape.drawLatest(@ctx, @bufferCtx)
+                if (@selectedShape? && shape.id == @selectedShape.id)
+                  createShape('SelectionBox', {
+                    shape: shape,
+                    handleSize: 0
+                  }).drawLatest(@ctx, @bufferCtx)
```

Comment:
Even though I think I understand why you did this (so the selection box would be at the same layer as the shape), I think I would prefer it if the selection tool handled this and just drew the box on top of everything by setting the selection box as its "shape in progress."